src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/PhpConfigReferenceDumpPass.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\Config\Definition\ArrayShapeGenerator;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Extension\ConfigurationExtensionInterface;
+use Symfony\Component\DependencyInjection\Loader\Configurator\AppReference;
+
+/**
+ * @internal
+ */
+class PhpConfigReferenceDumpPass implements CompilerPassInterface
+{
+    private const REFERENCE_TEMPLATE = <<<'EOPHP'
```

Comment:
Nit: PHPStorm has syntax highlighting and static analysis if you use the `PHP` HEREDOC ([doc](https://www.jetbrains.com/help/phpstorm/using-language-injections.html#inject-language-nowdoc-heredoc)).

```suggestion
    private const REFERENCE_TEMPLATE = <<<'PHP'
```